nwb_conversion_tools/utils/spike_interface.py

```diff
@@ -322,77 +322,68 @@ def add_electrodes(
+            if prop == "location":
+                location_map = ["rel_x", "rel_y", "rel_z"]
+                for prop_name_new, loc in zip(location_map, range(data.shape[1])):
+                    elec_columns[prop_name_new].update(description=prop_name_new, data=data[:, loc], index=False)
+            else:
+                if prop == "brain_area":
+                    prop = "location"
+                elif prop == "group":
+                    if "group_name" not in property_names:
+                        prop = "group_name"
+                    else:
+                        continue
+                index = isinstance(data[0], (list, np.ndarray))
+                elec_columns[prop].update(description=prop, data=data, index=index)
 
+    # Fill with provided custom descriptions
     for x in metadata["Ecephys"]["Electrodes"]:
-        elec_columns[x["name"]]["description"] = x["description"]
         if x["name"] not in list(elec_columns):
-            raise ValueError(f'"{x["name"]}" not a property of se object')
+            raise ValueError(f'"{x["name"]}" not a property of se object, set it first and rerun')
+        elec_columns[x["name"]]["description"] = x["description"]
 
-    # updating default arguments if electrodes table already present:
+    # Updating default arguments if electrodes table already present:
     default_updated = dict()
     if nwbfile.electrodes is not None:
         for colname in nwbfile.electrodes.colnames:
             if colname != "group":
                 samp_data = nwbfile.electrodes[colname].data[0]
-                default_datatype = [
-                    proptype for proptype in channel_property_defaults if isinstance(samp_data, proptype)
-                ][0]
-                default_updated.update({colname: channel_property_defaults[default_datatype]})
+                default_datatype = [proptype for proptype in property_default_types if isinstance(samp_data, proptype)][
+                    0
+                ]
+                default_updated.update({colname: property_default_types[default_datatype]})
     default_updated.update(defaults)
 
+    # Re-shape data to have as many entries as channels are available in the table
     for name, des_dict in elec_columns.items():
         des_args = dict(des_dict)
         if name not in default_updated:
             if nwbfile.electrodes is None:
                 nwbfile.add_electrode_column(name=name, description=des_args["description"], index=des_args["index"])
             else:
-                combine_data = [channel_property_defaults[found_property_types[name]]] * len(nwbfile.electrodes.id)
-                des_args["data"] = combine_data + des_args["data"]
+                samp_data = des_dict["data"][0]
+                data_type_found = [proptype for proptype in property_default_types if isinstance(samp_data, proptype)][
+                    0
+                ]
```

Comment:
```suggestion
                data_type_found = [proptype for proptype in property_default_types if isinstance(des_dict["data"][0], proptype)][
                    0
                ]
```
`samp_data` wasn't being used again after this, right?

Reply:
No.
The reason to define a name for it is that I found that many things were done inside the list comprehension and I wanted to make it more clear through naming in the variables that the operation that is happening there is that "the data type (name of the final variable) is found by the sample data the default property types to the sample data" 

The same mechanism is used in the loop before that: 
https://github.com/catalystneuro/nwb-conversion-tools/blob/86a2bfb32236c6aab0215b35cdb7ceceafa10df1/nwb_conversion_tools/utils/spike_interface.py#L358-L368

Reply:
Yeah I hear you, we run into this from time to time when we're doing a list search that only returns a single element. I'd leave as is and refactor later; for reference the better way to do it is an explicit `for` loop like

```python
for proptype in property_default_types:
    if isinstance(nwbfile.electrodes[colname].data[0] , proptype)]:
        default_datatype = proptype
        break
```
The break command will then exit the `for` loop as soon as the `proptype` is found, so more efficient too.